ui/pages/create-account/import-account/import-account.js

```diff
@@ -0,0 +1,159 @@
+  const { navigateToMostRecentOverviewPage } = useRouting();
+
+  const menuItems = [t('privateKey'), t('jsonFile')];
+
+  const [type, setType] = useState(menuItems[0]);
+
+  function importAccount(strategy, importArgs) {
+    const loadingMessage = getLoadingMessage(strategy);
+
+    dispatch(actions.importNewAccount(strategy, importArgs, loadingMessage))
+      .then(({ selectedAddress }) => {
+        if (selectedAddress) {
+          trackImportEvent(strategy, true);
+          dispatch(actions.hideWarning());
+          navigateToMostRecentOverviewPage();
+        } else {
+          dispatch(actions.displayWarning(t('importAccountError')));
+          trackImportEvent(strategy, false);
+        }
+      })
+      .catch((error) => translateWarning(error.message));
+  }
+
+  function trackImportEvent(strategy, wasSuccessful) {
+    const accountImportType =
+      strategy === 'Private Key'
+        ? EVENT.ACCOUNT_IMPORT_TYPES.PRIVATE_KEY
+        : EVENT.ACCOUNT_IMPORT_TYPES.JSON;
+
+    const event = wasSuccessful
+      ? EVENT_NAMES.ACCOUNT_ADDED
+      : EVENT_NAMES.ACCOUNT_ADD_FAILED;
+
+    trackEvent({
+      category: EVENT.CATEGORIES.ACCOUNTS,
+      event,
+      properties: {
+        account_type: EVENT.ACCOUNT_TYPES.IMPORTED,
+        account_import_type: accountImportType,
+      },
+    });
+  }
+
+  function getLoadingMessage(strategy) {
+    if (strategy === 'JSON File') {
+      return (
+        <Text width={BLOCK_SIZES.THREE_FOURTHS} fontWeight={FONT_WEIGHT.BOLD}>
+          <br />
+          {t('importAccountJsonLoading1')}
+          <br />
+          <br />
+          {t('importAccountJsonLoading2')}
+        </Text>
+      );
+    }
+
+    return '';
+  }
+
+  /**
+   * @param {string} message - an Error/Warning message caught in importAccount()
+   * This function receives a message that is a string like:
+   * `t('importAccountErrorNotHexadecimal')`
+   * `t('importAccountErrorIsSRP')`
+   * `t('importAccountErrorNotAValidPrivateKey')`
+   * and feeds it through useI18nContext
+   */
+  function translateWarning(message) {
+    if (message && !message.startsWith('t(')) {
+      // This is just a normal error message
+      dispatch(actions.displayWarning(message));
+    } else {
+      // This is an error message in a form like
+      // `t('importAccountErrorNotHexadecimal')`
+      // so slice off the first 3 chars and last 2 chars, and feed to i18n
+      dispatch(actions.displayWarning(t(message.slice(3, -2))));
+    }
+  }
+
+  function PrivateKeyOrJson() {
+    switch (type) {
+      case menuItems[0]:
+        return <PrivateKeyImportView importAccountFunc={importAccount} />;
+      case menuItems[1]:
+      default:
+        return <JsonImportView importAccountFunc={importAccount} />;
+    }
+  }
+
+  return (
+    <>
+      <Box
+        padding={4}
+        style={{ borderWidth: '0px 0px 1px 0px' }} // There is no way to do just a bottom border in the Design System
```

Comment:
I think unfortunately for any additional styles we want to use a scss file and css? I think it is slightly more performant that inline styles as well.